src/main/java/carpet/script/api/Scoreboards.java

```diff
@@ -147,5 +154,295 @@ public static void apply(Expression expression)
             scoreboard.setObjectiveSlot(slot, objective);
             return (_c, _t) -> new NumericValue(slot);
         });
+
+        expression.addLazyFunction("team_list", -1, (c, t, lv) ->
+        {
+            if(lv.size() > 1) throw new InternalExpressionException("'team_list' requires zero or one parameters");
+            CarpetContext cc = (CarpetContext)c;
+            ServerScoreboard scoreboard = cc.s.getMinecraftServer().getScoreboard();
+            if(lv.size() == 0) {
+                Value ret = ListValue.wrap(scoreboard.getTeamNames().stream().map(StringValue::new).collect(Collectors.toList()));
+                return (_c, _t) -> ret;
+            } else if(lv.size() == 1) {
+                Value teamVal = lv.get(0).evalValue(c);
+                if(teamVal instanceof StringValue) {
```

Comment:
I suggest not being that strict here. use .getString() on any value. If team '123' exists, that's fine I would say.